src/hd-account.js

```diff
@@ -12,25 +12,24 @@ var constants = require('./constants');
 // HDAccount Class
 
 function HDAccount (object) {
-  var self = this;
   var obj = object || {};
   obj.cache = obj.cache || {};
-  obj.address_labels = obj.address_labels || [];
   // serializable data
   this._label = obj.label;
   this._archived = obj.archived || false;
   this._xpriv = obj.xpriv;
   this._xpub = obj.xpub;
   this._network = obj.network || Bitcoin.networks.bitcoin;
 
-  this._address_labels = [];
-  obj.address_labels.map(function (e) { self._address_labels[e.index] = e.label; });
+  // Prevent deleting address_labels field when saving wallet:
+  // * backwards compatibility with mobile (we'll keep one entry for the highest label)
+  // * if user has 2nd password enabled and doesn't enter it during migration step
+  this._address_labels_backup = obj.address_labels;
 
   // computed properties
   this._keyRing = new KeyRing(obj.xpub, obj.cache);
-  this._receiveIndex = 0;
   // The highest receive index with transactions, as returned by the server:
-  this._lastUsedReceiveIndex = 0;
+  this._lastUsedReceiveIndex = -1;
```

Comment:
I could use `null` instead for clarity and refactor the code that determines the receiveIndex.